src/services/users/users.hooks.js

```diff
@@ -2,33 +2,36 @@ const { authenticate } = require('@feathersjs/authentication').hooks;
 const { hashPassword, protect } = require('@feathersjs/authentication-local').hooks;
 const {
   iff,
+  isNot,
   isProvider,
   preventChanges,
   disallow,
 } = require('feathers-hooks-common');
 const verifyHooks = require('feathers-authentication-management').hooks;
+const { verifyRegisterToken, hasVerifyToken } = require('../../hooks/userhooks');
 const permission = require('../../hooks/permission');
 const accountService = require('../authmanagement/notifier');
 
 
 module.exports = {
   before: {
-    all: [authenticate('jwt')],
-    find: [permission({ roles: ['admin'] })],
-    get: [permission({ roles: ['admin', 'manager', 'coach'] })],
+    all: [],
+    find: [
+      iff(isProvider('external'),
+        iff(isNot(hasVerifyToken()),
+          authenticate('jwt'),
+          permission({ roles: ['admin'] }))),
+    ],
+    get: [authenticate('jwt'), permission({ roles: ['admin', 'manager', 'coach'] })],
     create: [
       hashPassword(),
-      permission({ roles: ['admin'] }),
       disallow('external'),
-      iff(
-        isProvider('external'),
-        verifyHooks.addVerification(),
```

Comment:
We need this to add the isVerified field don't we?

Reply:
That iff statement should never be reach with the disallow external right above it. So we would need to remove one or the other, your choice lel